avocado/utils/vmimage.py

```diff
@@ -739,14 +900,24 @@ def get_best_provider(name=None, version=None, build=None, arch=None):
         if name == "fedora" and arch in ("ppc64", "ppc64le", "s390x"):
             name = "fedorasecondary"
 
+    last_error = None
     for provider in IMAGE_PROVIDERS:
         if name is None or name == provider.name.lower():
             try:
                 return provider(**provider_args)
             except ImageProviderError as e:
                 LOG.debug(e)
+                last_error = e
```

Comment:
I don't think that raising only the `last_error` is a right way to do it. What happens when the second error, less important than the first error? The user will get misleading information. IMO, if we want to raise the error, we should raise all of them or just the first one and then stop.